src/hnsw/build.c

```diff
@@ -333,13 +333,16 @@ static void BuildIndex(
         assert(error == NULL);
         elog(INFO, "done loading usearch index");
 
-        // todo determine how we can set index rd_options
-        // HnswOptions *opts = (HnswOptions *)index->rd_options;
-        // opts->m = usearch_connectivity(buildstate->usearch_index, &error);
-        // assert(error == NULL);
-        // opts->dims = usearch_dimensions(buildstate->usearch_index, &error);
-        // assert(error == NULL);
-        // todo determine how to get ef construction and ef search params
+        opts.connectivity = usearch_connectivity(buildstate->usearch_index, &error);
+        assert(error == NULL);
+        opts.dimensions = usearch_dimensions(buildstate->usearch_index, &error);
+        assert(error == NULL);
+        opts.expansion_add = usearch_expansion_add(buildstate->usearch_index, &error);
```

Comment:
Could all of these be retrieved from `usearch_metadata`?

Reply:
There should be done some changes in usearch to return this data from metadata but yes technically it is possible.

Reply:
I have added this params to `usearch_metadata` in [this](https://github.com/Ngalstyan4/usearch/pull/11) PR